### DISCUSSION TITLE
支持班主任发通知

### DISCUSSION THREAD

**Discussion by pdog18** [Class-Widgets/Class-Widgets#294] [Feature]
### 检查清单

- [X] 我已更新到最新版 ![最新的版本号](https://img.shields.io/github/v/release/RinLit-233-shiroko/Class-Widgets?include_releases&style=flat-square&label=) ，并看过[最新提交](https://github.com/RinLit-233-shiroko/Class-Widgets/commits/main/)，确认我想要的功能还没有实现。
- [X] 我已在 [Issues](https://github.com/RinLit-233-shiroko/Class-Widgets/issues?q=label%3ABug) 中检索，确认这一 Bug 未被提交过。
- [X] 我认为我想要的功能是用户广泛需要的，插件不能替代的，且没有超出 **软件本来** 的开发目标，而非 **添加与课表及教学辅助无关的内容** 。
- [X] 我已知晓由于开发者时间精力有限，我提交的功能请求可能需要比较久的时间才能实现。

### 背景与动机

这个软件非常好用，我们学校推广时收到了老师们的一致好评，都很开心有这么好的软件。同时也希望这个软件对于学校管理方面可以发挥出更多的作用。

### 想要实现或优化的功能

希望增加班主任给班级发布通知的功能，以往需要去班级里面通知，如果可以直接通过手机发布就好了。
感觉这个功能因为涉及到使用服务器资源，可以考虑做成收费功能。只要价格不是特别高，甚至不需要找学校采购，老师自己都愿意付费。毕竟给老师们带来了方便。

### 最后一步

- [X] 我认为上述的描述已经足以详细，以便开发人员理解您想要的功能。如果我的 issue 没有按照上述的要求填写，可能会被无条件关闭。

### 不要乱下载别人发的文件，特别是名字是fix.zip的文件。您是否遇见过这样的文件？

是

**Comment by MKStoler4096** [comment]
感觉像是ci尚未发布的集控（）但是cw看起来**没有单独的“通知”逻辑**，这是个问题。

也不用收费，py开服务器很简单，根本不用花钱组，出个部署教程就好了

**Comment by Leozhu2010** [comment]
希沃集控：  
不过考虑到还有很多学校没买集控服务/用的别的厂商的一体机，似乎还挺合理的（笑）

**Comment by Leozhu2010** [comment]
> 感觉像是ci尚未发布的集控（）但是cw看起来**没有单独的“通知”逻辑**，这是个问题。
> 
> 也不用收费，py开服务器很简单，根本不用花钱组，出个部署教程就好了

直接用上下课提醒弹窗那一套窗口（划

**Comment by pdog18** [comment]
我感觉这个功能可玩性还是挺高的。比如我们班主任有时候会「召唤」某个同学到办公室（发布任务，笑）。
那通过这个发出就可以了，但是为了避免同学们没有留意到，而班主任又不知道。所以这个通知在发布的时候应该提供一个选项，是否开启「已读确认」。开启这个功能时，一体机上将会有一个确认按钮提供给班上的同学（这个按钮应该还要做防误触的概念）

如果有这个已读确认，那么班主任给同学们发送「下节语文改成体育课」这种不需要同学们到办公室的通知，就可以让老师们知道，同学们是不是已经看到这个通知了？

**Comment by MKStoler4096** [comment]
这个想法很好，估计专门要写集控服务的ci都要头大（）🤣

**Comment by Welsonpeaches** [comment]
吸我疾控+有钱的学校解决一切
